sqlconnect/internal/snowflake/legacy_mappings.go

```diff
@@ -44,6 +44,7 @@ func legacyColumnTypeMapper(columnType base.ColumnType) string {
 		"TIMESTAMP_LTZ":    "datetime",
 		"TIMESTAMP_TZ":     "datetime",
 		"VARIANT":          "json",
+		"ARRAY":            "array",
```

Comment:
Can we adapt normal mappings as well to return `array` as well instead of `json`?
I believe we would need to adapt both snowflake & bigquery

Reply:
why are we using `json` for databricks legacy?

Reply:
Will change normal mappings for snowflake. wasn't sure if its safe to change, in case anyone else is using it that's why didn't touch
In bigquery we are no really supporting complex types in retl itself. we just convert any complex type to string. That would need more thought
Also, the specific feature is only for snowflake to unblock a prospect.

Reply:
noone uses non-legacy mappings

Reply:
thanks, please make sure tests are green too!